packages/workspace/src/command-line/run-tasks/run-command.ts

```diff
@@ -0,0 +1,205 @@
+import { DependencyGraph, NxJson, ProjectNode } from '../shared';
+import {
+  AffectedEventType,
+  Task,
+  TaskCompleteEvent,
+  TasksRunner
+} from '../../tasks-runner/tasks-runner';
+import { defaultTasksRunner } from '../../tasks-runner/default-tasks-runner';
+import { isRelativePath } from '../../utils/fileutils';
+import { join } from 'path';
+import { appRootPath } from '../../utils/app-root';
+import { DefaultReporter, ReporterArgs } from './default-reporter';
+import {
+  Arguments,
+  Environment,
+  projectHasTargetAndConfiguration
+} from './utils';
+import * as yargs from 'yargs';
+
+interface TasksMap {
+  [projectName: string]: { [targetName: string]: Task };
+}
+
+type RunArgs = yargs.Arguments & ReporterArgs;
+
+export function runCommand<T extends RunArgs>(
+  projects: ProjectNode[],
+  projectGraph: DependencyGraph,
+  { nxArgs, overrides, targetArgs }: Arguments<T>,
+  { nxJson, workspace }: Environment
+) {
+  const reporter = new DefaultReporter();
+  reporter.beforeRun(projects.map(p => p.name), nxArgs, overrides);
+
+  const tasks: Task[] = projects.map(project =>
+    createTask({
+      project,
+      target: nxArgs.target,
+      configuration: nxArgs.configuration,
+      overrides: overrides
+    })
+  );
+
+  const tasksMap: TasksMap = {};
+  projects.forEach(project => {
```

Comment:
TaskMap is a set of all possible tasks, so it cannot be derived from the list of projects you are currently running.

Reply:
In the current version is done like this:

Object.entries(affectedMetadata.dependencyGraph.projects)

So we generate a task map from the list of all possible projects